src/internals/ir/callGraph.ts

```diff
@@ -0,0 +1,193 @@
+import { TactASTStore } from "./astStore";
+import { IdxGenerator } from "./indices";
+import {
+  AstFunctionDef,
+  AstReceiver,
+  AstContractInit,
+  AstStatement,
+  AstStaticCall,
+  AstMethodCall,
+  AstExpression,
+  AstStructFieldInitializer,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+class CGEdge {
+  public idx: number;
+  constructor(
+    public src: number,
+    public dst: number,
+  ) {
+    this.idx = IdxGenerator.next("edge");
+  }
+}
+
+class CGNode {
+  public idx: number;
```

Comment:
I suggest creating type aliases in order to avoid confusion here and afterward:
```typescript
export type CGNodeIdx = number;
export type CGNodeAstId = number;
// ... the same for CGEdge
```